src/main/resources/templates/scala-lang/scalaClassGraphqlResponseProjection.ftl

```diff
@@ -26,16 +29,18 @@ import java.util.Objects
 </#list>
 class ${className} extends GraphQLResponseProjection {
 
-<#if fields?has_content>
-    override def all$(): ${className} = all$(${responseProjectionMaxDepth})
+<#if fields?has_content && generateAllMethodInProjection>
+    val projectionDepthOnFields = new HashMap[String, Int]
```

Comment:
use `private final  lazy val projectionDepthOnFields = new HashMap[String, Int]`

Reply:
Thanks for the suggestion!